tasks/Linux/fetch/dragonwell8-fallback.yml

```diff
@@ -0,0 +1,49 @@
+---
+- name: Check requested major version for Alibaba Dragonwell
+  fail:
+    msg: "Alibaba Dragonwell corresponds to OpenJDK 8 only"
+  when: java_major_version != 8
+
+- name: Check requested architecture for Alibaba Dragonwell
+  fail:
+    msg: "Alibaba Dragonwell JDK currently supports Linux/x86_64 platform only"
+  when: java_arch != "x64"
+
+- name: Prepare GitHub API release tag
+  set_fact:
+    sub_uri: >-
+      {{ ((java_minor_version != '*')
+        | ternary('tags/v{{ java_major_version }}.{{ java_minor_version }}-GA', 'latest')) }}
+
+- name: Fetch Dragonwell version by tag
+  uri:
+    url: "{{ github_api_page }}/repos/alibaba/dragonwell8/releases/{{ sub_uri }}"
+    return_content: true
+    body_format: json
+    status_code: [200, 404] # Not found case we are handling below
+  register: response
+
+- name: Exit if Dragonwell version is not found
+  fail:
+    msg: 'Dragonwell version {{ java_major_version }}.{{ java_minor_version }} not found'
+  when: response.status == 404
+
+# Transform Release name to Asset name
+# Release name sample: Alibaba Dragonwell 8.1.1-GA
+# Asset name sample: Alibaba_Dragonwell_8.1.1-GA_Linux_x64.tar.gz
+- name: Prepare release asset name
+  set_fact:
+    asset_name: "{{ (response.json.name | regex_replace(' ', '_')) + '_Linux_x64.tar.gz' }}"
+
+- name: Get release URL
+  set_fact:
+    release_url: "{{ response.json | json_query(\"assets[?name=='\" + asset_name + \"'].browser_download_url\") }}"
+
+- name: Download artifact from {{ release_url[0] }}
```

Comment:
How about checksum?

Reply:
GitHub API doesn't provide checksum for release assets. Check it yourself:
"assets": [
    {
      "url": "https://api.github.com/repos/alibaba/dragonwell8/releases/assets/13383833",
      "id": 13383833,
      "node_id": "MDEyOlJlbGVhc2VBc3NldDEzMzgzODMz",
      "name": "Alibaba_Dragonwell_8.0.0-GA_Linux_x64.tar.gz",
      "label": null,
      "uploader": {
        "login": "yfxhust",
        "id": 5487895,
        "node_id": "MDQ6VXNlcjU0ODc4OTU=",
        "avatar_url": "https://avatars1.githubusercontent.com/u/5487895?v=4",
      ...
      },
      "content_type": "application/x-gzip",
      "state": "uploaded",
      "size": 105990732,
      "download_count": 3164,
      "created_at": "2019-06-25T04:01:25Z",
      "updated_at": "2019-06-25T04:02:29Z",
      "browser_download_url": "https://github.com/alibaba/dragonwell8/releases/download/v8.0.0-GA/Alibaba_Dragonwell_8.0.0-GA_Linux_x64.tar.gz"
    },